check-style

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Unset variable would be a sign of programmer error. We are not using '-e' in
+# this script as we'd like to handle these cases ourselves where relevant, i.e.,
+# allow more than one code check failure per run.
+set -u
+
+# Work from the git root. This is important to help some tools pick up the
+# correct configuration.
+git_root=$(git rev-parse --show-toplevel)
+cd "${git_root}"
+
+# Get all files under version control.
+affected_files=$(git ls-tree --full-tree --name-only -r HEAD)
+
+# Cummulative status code for the script. Updated through 'run_check' when
+# running checks. A status code creater than 0 will indicate that one or more
+# checks failed.
+status_code=0
+
+# Run a code check command and update the cummulative error code.
+run_check() {
```

Comment:
This file could use a comment explaining how to use it, most notably the expected arguments and what it does with them. It seems like we're supposed to pass a program to run?